internal/storage/serde.go

```diff
@@ -1027,8 +1134,9 @@ func NewBinlogSerializeWriter(schema *schemapb.CollectionSchema, partitionID, se
 			arrays[i] = builder.NewArray()
 			builder.Release()
 			fields[i] = arrow.Field{
-				Name: strconv.Itoa(int(fid)),
-				Type: arrays[i].DataType(),
+				Name:     strconv.Itoa(int(fid)),
+				Type:     arrays[i].DataType(),
+				Nullable: true, // No nullable check here.
```

Comment:
why no nullable check here? it seem ok to get nullable in collection schema.

Reply:
There are two reasons why there is no nullable check:
1. In compact operations, it is safe to suppose the compacting segments are valid. The nullable checks have been done at the insert time.
2. The schema evolution feature *requires* nullable for almost all columns, making the schema property “not-nullable” useless. I suppose “not-nullable” may useful only in compile-time, e.g., in literal fields for constant folding optimizations.